src/sinks/util/buffer/compression.rs

```diff
@@ -159,11 +186,29 @@ impl<'de> de::Deserialize<'de> for Compression {
                     },
                     "gzip" => Ok(Compression::Gzip(level.unwrap_or_default())),
                     "zlib" => Ok(Compression::Zlib(level.unwrap_or_default())),
+                    "zstd" => Ok(Compression::Zstd(level.unwrap_or_default())),
                     algorithm => Err(de::Error::unknown_variant(
                         algorithm,
-                        &["none", "gzip", "zlib"],
+                        &["none", "gzip", "zlib", "zstd"],
                     )),
-                }
+                }?;
+
+                match compression.compression_level() {
+                    CompressionLevel::Val(level) => {
+                        let max_level = compression.max_compression_level_val();
+                        if level > max_level {
+                            let msg = std::format!(
+                                "invalid value `{}`, expected value in range [0, {}]",
```

Comment:
```suggestion
                                "Invalid value `{}`, expected value in range [0, {}].",
```

Not sure about this, but I think [the style guide applies here](https://github.com/vectordotdev/vector/blob/master/STYLE.md#writing-a-good-log-message)?

Reply:
Here I tried to stay close to serde's error messages style. After fixing the error will be like that
`Invalid value `10`, expected value in range [0, 9]. at line 1 column 34`, which doesn't fit to serde's style.